File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums) {
        let l = 1;
        for(let r = 1; r < nums.length; r++) {
            if(nums[r] != nums[r-1]) {
                nums[l++] = nums[r];
            }
        }
        return l;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        boolean flag = false;
        int k = 0;
        for (int i = 1; i < nums.length - k; i++) {
            int diff = nums[i] - nums[i-1];
            if(diff == 0) {
                if (flag) {
                    nums = shiftArr(nums, i);
                    k++;
                    i--;
                } else {
                    flag = true;
                }
            } else if(flag) {
                flag = false;
            }
        }
        return nums.length - k;
    }

    private int[] shiftArr(int[] arr, int k) {
        while (k + 1 < arr.length) {
            int temp = arr[k];
            arr[k] = arr[k+1];
            arr[k + 1] = temp;
            k++;
        }
        return arr;
    } 
}